### PR TITLE
add general sql layer and mysql support

### PR DESCRIPTION
**Pull request by sound2gd**
adds support for general sql support. including

- `kit-sql-general` - General sql layer, just contains connection pooling (via [hikari-cp](https://github.com/tomekw/hikari-cp)) and jdbc wrapper (via [next.jdbc](https://github.com/seancorfield/next-jdbc)).
- `kit-mysql` - JDBC Driver working with MySQL8+

daily sql stack here: MySQL8, hikaricp for connection pooling, next-jdbc for DB sql talking, honeysql for sql manipulating.

### CODE REVIEW

**Comment by nikolap** [comment]
Thanks for the PR!

I'll have more time on the weekend to review this, but some initial thoughts:

- `kit-mysql` probably doesn't need to be a library, since all it is is pulling in a dependency. It is clearer to have this dependency explicit in your deps.edn rather than transient through a library if that library does nothing else. The difference between with `kit-postgres` library is it has are [extensions to some next-jdbc protocols](https://github.com/kit-clj/kit/blob/master/libs/kit-postgres/src/kit/edge/db/postgres.clj#L36)
- Decoupling the migrations from the general sql edge might be useful... We probably don't need to tie these two together, i.e. right now `kit-sql` bundles both [conman](https://github.com/luminus-framework/conman) and [migratus](https://github.com/yogthos/migratus) together, but it probably doesn't need to... Perhaps we can keep conman, since you can still use it for the convenience when integrating with hikari-cp, but you don't need to explicitly use the Hugsql binding generation (the `query-fn` integrant component can just be removed).

**Comment by sound2gd** [comment]
>  It is clearer to have this dependency explicit in your deps.edn rather than transient through a library if that library does nothing else.
agree. and will add some java-8 types extension for mysql.

> Perhaps we can keep conman, since you can still use it for the convenience when integrating with hikari-cp, but you don't need to explicitly use the Hugsql binding generation
conman bundled with a lot unused dependencies. such as hugsql. 

A general sql layer should ship with  connection pooling and jdbc wrapper.
 conman and hugsql shoud live above this general layer, maybe as "sql manipulation layer".

**Comment by yogthos** [comment]
I agree with the rationale regarding conman, we also might want to look at updating the [sqlite module](https://github.com/kit-clj/modules/blob/master/sqlite/config.edn) to match the changes as well as adding mysql and postgres modules.

**Comment by nikolap** [comment]
Makes sense. How does this approach sound @yogthos @sound2gd :

- Separate current `kit-sql` library into four, with `kit-sql` remaining as a means of backwards compatibility:
  - `kit-sql` pulls in `kit-sql-conman` and pulls in `kit-migratus` as a dependency. It doesn't do anything fancy beyond being a wrapper for both. This is to avoid breaking changes for anyone relying on the `kit-sql` library to provide both.
  - `kit-sql-conman` is the sql library with conman/hugsql and next-jdbc.
  - `kit-sql-hikari` is the sql library with hikari-cp and next-jdbc (i.e. from @sound2gd's PR as `kit-sql-general`) 
  - `kit-sql-migratus` adds migratus. This decouples migrations from the core sql library. Can be handy if people want to pull in their own migrations tool.
- The lein-template will not generate new projects with `kit-sql`, but instead explicitly call whichever ones are necessary. We should figure out how the profile rules will apply for code generation / combinations.

Separately: with 21c624fd0a463844c7493fcd0321fe36cbd48a01, it makes sense now have the `kit-mysql` library as well.

**Comment by sound2gd** [comment]
@nikolap sounds reasonable, will do some coding later this week

File: .gitignore [hunk 1]
```diff
@@ -14,4 +14,6 @@ pom.xml.asc
 .idea/
 .cpcache/
 .DS_Store
-.nrepl-port
\ No newline at end of file
+.nrepl-port
+.lsp
+.clj-kondo/
```
Comment (NoahTheDuke) [inline .gitignore:19]: Should this be `.clj-kondo/.cache/`? So any changes to `.clj-kondo/config.edn` aren't ignored

**Comment by nikolap** [comment]
Resolved with https://github.com/kit-clj/kit/pull/25

**Pull request merged**